### ISSUE TITLE
Add a global dirtycheck event that is shared by all plugins

### ISSUE THREAD

**Issue by csuwildcat** [MicrosoftDX/Vorlonjs#97]
To prevent the pattern of every plugin assaulting performance to check for things in the target remote page, add a global dirtycheck event that they can all tap into.

**Comment by csuwildcat** [comment]
@deltakosh @gleborgne I was able to add a global event for content change (via dirty checking client content) that is broadcast to all plugins, but when I attempted to get latest the DOM Explorer plugin had radically changed in structure. The old dirty check is no longer needed after this addition, but I am unsure how to extricate it gracefully from the current code on master. I have a message with type `contentchanged` that is run on all clients, and broadcasts only when a change has occurred. All plugins receive the message event and the current content.

Any ideas on how to go about getting this in? I don't want to create issues for MCNext and others who are doing heavy modifications to the files. Let me know what you think.
